packages/flame_forge2d/lib/contact_callbacks.dart

```diff
@@ -1,93 +1,55 @@
-import 'package:forge2d/forge2d.dart';
-
-import 'body_component.dart';
-
-class ContactTypes<T1, T2> {
-  // If o1 is, or inherits from, T1 or T2
-  bool has(Object o1) => o1 is T1 || o1 is T2;
-  bool hasOne(Object o1, Object o2) => has(o1) || has(o2);
-
-  // Only makes sense to call with objects that you know is in [T1, T2]
-  bool inOrder(Object o1, Object o2) => o1 is T1 && o2 is T2;
-
-  // Remember that this is not symmetric, it checks if the types in `o1` and
-  // `o2` are the same or inherits from the types in `other`
-  bool match(Object o1, Object o2) =>
-      (o1 is T1 && o2 is T2) || (o2 is T1 && o1 is T2);
-}
-
-typedef ContactCallbackFun = void Function(Object a, Object b, Contact contact);
-
-abstract class ContactCallback<Type1, Type2> {
-  ContactTypes<Type1, Type2> types = ContactTypes<Type1, Type2>();
-
-  void begin(Type1 a, Type2 b, Contact contact) {}
-  void end(Type1 a, Type2 b, Contact contact) {}
-  void preSolve(Type1 a, Type2 b, Contact contact, Manifold oldManifold) {}
-  void postSolve(Type1 a, Type2 b, Contact contact, ContactImpulse impulse) {}
-}
-
-class ContactCallbacks extends ContactListener {
-  final List<ContactCallback> _callbacks = [];
-
-  void register(ContactCallback callback) {
-    _callbacks.add(callback);
+import 'flame_forge2d.dart';
+
+/// Used to listens to a [BodyComponent] contacts events.
+///
+/// Contacts events occur whenever two [Fixture] meet each other.
+///
+/// To react to contacts you should assign [ContactCallbacks] to a [Body]'s
+/// userData or/and to [Fixture]'s userData.
+/// {@macro flame_forge2d.world_contact_listener.algorithm}
+class ContactCallbacks {
+  /// Called when two [Fixture]s begin to overlap.
```

Comment:
```suggestion
  /// Called when two [Fixture]s start being in contact.
```

Reply:
Refer to https://github.com/flame-engine/flame/pull/1547#discussion_r856123357